csrc/aes_kwp.cpp

```diff
@@ -86,16 +82,11 @@ JNIEXPORT int JNICALL Java_com_amazon_corretto_crypto_provider_AesKeyWrapSpi_unw
         }
 
         jni_borrow inbuf(env, input, "input");
-        input.get_bytes(env, inbuf.data(), 0, input.len());
         jni_borrow outbuf(env, output, "output");
         size_t outlen;
         if (!AES_unwrap_key_padded(&aes_key, outbuf.data(), &outlen, input.len(), inbuf.data(), input.len())) {
```

Comment:
Fourth parameter still looks wrong. Shouldn't it be `outbuf.len()`?

Also, please initialize `outlen` to some value. Uninitialized variables are dangerous. (Yes, there are still some in the code-base, but we should fix them as we find them.)

Finally, it's better style to pull `len()` from the `jni_borrow` (as that's the object you're actually using) rather than from the `java_buffer`. There are edge-cases where borrows are not the same length as the underlying buffer. (At least I think there might be. Check the APIs to create a borrow.)

Reply:
> Fourth parameter still looks wrong. Shouldn't it be outbuf.len()?

yep, miss on my part. we should be bounding the max output buffer write size by the length of the output buffer. change made.

> Also, please initialize outlen to some value. Uninitialized variables are dangerous. (Yes, there are still some in the code-base, but we should fix them as we find them.)

makes sense. i considered initializing it to an invalid value (i.e. `-1`), but went with `0` as you suggest because that is also correct in the failure case (i.e. if the operation fails, 0 bytes will (should?) be written by AWS-LC).

> Finally, it's better style to ...

fair enough, that makes sense. if we derive `outbuf` from something other than the output array parameter down the line, then using the output array's length would risk an overrun.

> There are edge-cases where borrows are not the same length as the underlying buffer. (At least I think there might be. Check the APIs to create a borrow.)

hm, i'm not [seeing any today](https://github.com/corretto/amazon-corretto-crypto-provider/blob/develop/csrc/buffer.h#L326) but that doesn't mean there won't be tomorrow. point taken that we should be using the length value of the closest-at-hand buffer wrapping object that we have in a particular scope or function.